### ISSUE TITLE
[Bug] - Release is published even when there are no new changes

### ISSUE THREAD

**Issue by khemrajrathore** [joernio/joern#3823]
**Describe the bug**
We have scheduled releases every day, irrespective of any new changes, we end up releasing new version of Joern

**Expected behavior**
A newer version should be released only when there are changes which were not released in the previous version

**Example**
Joern version `2.0.152` is one such example which is identical to version `2.0.151`

**Comment by mpollmeier** [comment]
That's true, the only problem is to make sure that there's no loopholes where we end up without a release. 
E.g. currently we tag the release before we publish it - if the publishing fails for whatever reason (e.g. sonatype issues) then we'll end up not trying again. 

Irrespective, it's probably better to err on side of not producing too many useless releases, given that the process itself has been very solid recently.

**Comment by mpollmeier** [comment]
Even good ideas sometimes take their time :) 
https://github.com/joernio/joern/pull/5476
